AEPServices/Sources/storage/FileSystemNamedCollection.swift

```diff
@@ -0,0 +1,139 @@
+/*
+ Copyright 2023 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+*/
+
+
+import Foundation
+
+class FileSystemNamedCollection: NamedCollectionProcessing {
+    
+    private let queue = DispatchQueue(label: "FileSystemNamedCollection.barrierQueue")
+    private let adobeDirectory = "com.adobe.aep.datastore"
+    private var appGroupUrl: URL?
+    private let fileManager = FileManager.default
+    private let LOG_TAG = "FileSystemNamedCollection"
+    var appGroup: String?
+    
+    func setAppGroup(_ appGroup: String?) {
+        self.appGroup = appGroup
+        if let appGroup = appGroup {
+            self.appGroupUrl = fileManager.containerURL(forSecurityApplicationGroupIdentifier: appGroup)
+        }
+    }
+    
+    func getAppGroup() -> String? {
+        return appGroup
+    }
+    
+    func set(collectionName: String, key: String, value: Any?) {
+        guard let fileUrl = fileUrl(for: collectionName) else {
+            return
+        }
+        queue.sync {
+            if var dict = self.getDictFor(collectionName: collectionName) {
```

Comment:
Can you simplify this logic? 
var dict = self.getDictFor(collectionName: collectionName) ?? [:]
guard !dict.isEmpty  || value != nil else {
   return
}
dict[key] = value
// Persist to file

Reply:
Great feedback. 🙏